tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,420 @@
+        assert os.path.exists(tmp_model_path)
+        updated_model_config = MPAConfig.fromfile(tmp_model_path)
+        assert updated_model_config.model.backbone.type == backbone_type
+
+    @e2e_pytest_unit
+    def test_builder_build_model_config_abnormal_model_path(self):
+        """Raise ValueError with wrong model_config_path."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_model_config")
+        tmp_backbone_path = os.path.join(workspace_path, "backbone.yaml")
+        with pytest.raises(ValueError):
+            self.otx_builder.build_model_config("unexpected", tmp_backbone_path)
+
+    @e2e_pytest_unit
+    def test_builder_build_model_config_abnormal_backbone_path(self):
+        """Raise ValueError with wrong backbone_config_path."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_model_config")
+        tmp_model_path = os.path.join(workspace_path, "model.py")
+        with pytest.raises(ValueError):
+            self.otx_builder.build_model_config(tmp_model_path, "unexpected")
+
+    @e2e_pytest_unit
+    def test_builder_build_model_config_without_out_indices(self):
+        """Update model config without backbone's out_indices."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_model_config")
+        tmp_model_path = os.path.join(workspace_path, "model.py")
+        tmp_backbone_path = os.path.join(workspace_path, "backbone.yaml")
+        backbone_config = mmcv.load(tmp_backbone_path)
+        assert backbone_config["backbone"].pop("out_indices") == (3,)
+        mmcv.dump(backbone_config, tmp_backbone_path)
+        self.otx_builder.build_model_config(tmp_model_path, tmp_backbone_path)
+        updated_model_config = MPAConfig.fromfile(tmp_model_path)
+        assert "out_indices" in updated_model_config.model.backbone
+
+    @e2e_pytest_unit
+    def test_builder_build_model_config_backbone_pretrained(self):
+        """Update model config with backbone's pretrained path."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_model_config")
+        tmp_model_path = os.path.join(workspace_path, "model.py")
+        tmp_backbone_path = os.path.join(workspace_path, "backbone.yaml")
+        backbone_config = mmcv.load(tmp_backbone_path)
+        expected_pretrained = "pretrained.path"
+        backbone_config["backbone"]["pretrained"] = expected_pretrained
+        mmcv.dump(backbone_config, tmp_backbone_path)
+        self.otx_builder.build_model_config(tmp_model_path, tmp_backbone_path)
+        updated_model_config = MPAConfig.fromfile(tmp_model_path)
+        assert updated_model_config.model.pretrained == expected_pretrained
+
+
+class MockBackbone(nn.Module):
+    def __init__(self):
+        super(MockBackbone, self).__init__()
+
+    def forward(self, x):
+        return x
+
+
+class MockRegistry(Registry):
+    def __init__(self, name, parent=None, scope=None):
```

Comment:
Can you provide type hints for these args to improve readability please?

Reply:
Added type hints.